src/EditorFeatures/Core/Implementation/CodeActions/CodeActionEditHandlerService.cs

```diff
@@ -147,9 +146,7 @@ internal class CodeActionEditHandlerService : ForegroundThreadAffinitizedObject,
             var singleChangedDocument = TryGetSingleChangedText(oldSolution, operations);
             if (singleChangedDocument != null)
             {
-                // ConfigureAwait(true) so we come back to the same thread as 
-                // we do all application on the UI thread.
-                var text = await singleChangedDocument.GetTextAsync(cancellationToken).ConfigureAwait(true);
+                var text = singleChangedDocument.GetTextSynchronously(cancellationToken);
```

Comment:
Chatting with @ivanbasov I suspect the flakiness is _usually_ GetTextAsync() would respond immediately, and it should always respond immediately in the open file case. So if you're doing a refactoring that only touches another file (like generate method...) and that one was closed and happened to get kicked out, boom.

Reply:
The theory also matches with my long-standing belief that we always had a bug here somewhere... :smile:

Reply:
> and that one was closed and happened to get kicked out, boom.

What does that mean?  what's the 'boom' part?

Reply:
You deadlock. Which I guess is anything but a "boom".